src/Illuminate/Database/Query/Grammars/Grammar.php

```diff
@@ -1119,6 +1119,8 @@ protected function wrapJsonFieldAndPath($column)
      */
     protected function wrapJsonPath($value, $delimiter = '->')
     {
+        $value = preg_replace("/([\\\\]+)?\\'/", "\\'", $value);
```

Comment:
With `preg_replace("/(\\\\)*'/", ...)`, the tests still pass. Is there a difference?

Reply:
You sure? Because it won't match `\'`, this would cause only the `'` to be escaped. Resulting in `\\'` as the end result, which in turn allows for the attack.

Reply:
Don't the tests check this?

Reply:
They do: https://github.com/laravel/framework/pull/28160/files#diff-132870d3cbe60fc402aa1a0a8b47387cR2266

Reply:
Same thoughts as staudenmeir, `preg_replace("/\\\\*'/", ...)` appears to be a strictly equivalent code. But simpler thus less error-prone.

I guess you got a bit confused between PHP and regex escaping ;)

Reply:
Describing the code:
* replace: quote preceded by 0 or more backslashes
* with: quote preceded by one backslash